io/src/main/java/com/github/jorgecastillo/kotlinandroid/io/algebras/ui/Presentation.kt

```diff
@@ -41,12 +40,14 @@ object Presentation {
     fun onHeroListItemClick(ctx: Context, heroId: String): IO<Unit> =
             Navigation.goToHeroDetailsPage(ctx, heroId)
 
-    private fun displayErrors(view: SuperHeroesView, t: Throwable): IO<Unit> = fx {
-        !effect {
-            when (CharacterError.fromThrowable(t)) {
-                is CharacterError.NotFoundError -> view.showNotFoundError()
-                is CharacterError.UnknownServerError -> view.showGenericError()
-                is CharacterError.AuthenticationError -> view.showAuthenticationError()
+    private fun displayErrors(view: SuperHeroesView, t: Throwable): IO<Unit> = unsafe {
```

Comment:
Why is unsafe necessary here?

Reply:
It isn't, still not removed. This is wip!